django/db/models/query.py

```diff
@@ -2194,7 +2198,10 @@ def model_fields(self):
         converter = connections[self.db].introspection.identifier_converter
         model_fields = {}
         for field in self.model._meta.fields:
-            model_fields[converter(field.column)] = field
+            # Fields with None "column" should be ignored
+            # (e.g. CompositePrimaryKey).
+            if field.column:
+                model_fields[converter(field.column)] = field
```

Comment:
How do you feel about making this a comprehension dict?
```python
        return {
            converter(field.column): field
            for field in self.model._meta.fields
            # Fields with None "column" should be ignored
            # (e.g. CompositePrimaryKey).
            if field.column
        }
```

Reply:
Sure, if you think it's worth doing. Updated.